src/Mutator/Operator/SpreadRemoval.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Mutator\Operator;
+
+use Infection\Mutator\Definition;
+use Infection\Mutator\GetMutatorName;
+use Infection\Mutator\Mutator;
+use Infection\Mutator\MutatorCategory;
+use PhpParser\Node;
+
+/**
+ * @internal
+ *
+ * @implements Mutator<Node\Expr\ArrayItem>
+ */
+final class SpreadRemoval implements Mutator
+{
+    use GetMutatorName;
+
+    public static function getDefinition(): ?Definition
+    {
+        return new Definition(
+            <<<'TXT'
+Removes a spread operator in an array expression. For example:
+
+```php
+$x = [...$collection, 4, 5];
+```
+
+Will be mutated to:
+
+```php
+$x = [$collection, 4, 5];
+```
+TXT
+            ,
+            MutatorCategory::SEMANTIC_REDUCTION,
+            null,
+            <<<'DIFF'
+- $x = [...$collection, 4, 5];
++ $x = [$collection, 4, 5];
```

Comment:
This particular mutator will be killed by static analyzers, which makes it useless.

Please, see the following Psalm examples:

* `$x = [...$collection, 4, 5];` - https://psalm.dev/r/4b159a69f1 - no issues
* `$x = [$collection, 4, 5];` - https://psalm.dev/r/9e9b78e2e1 Psalm error - `11:12 - Argument 1 of acceptsInt expects int, array{1, 2, 3} provided`

in other words, if we are using for example https://github.com/Roave/infection-static-analysis-plugin - this mutant will always be killed, resulting in wasted resources.

So I would vote to not do this mutation. Does it make sense?

Reply:
interesting point, I'm familiar with that plugin, however, in your example we are making an assumption: we assume that the array is a list of a single type, which might not be the case. There are many situations where an array will have mixed types and this mutant becomes valuable. 

Here's the simplest example, based on yours: https://psalm.dev/r/6ba915ee93. I've added the assertions at the end to show that the first element of each new array is not the same.

There will be other cases where the union type isn't even mentioned because an array can have many different types, so the developer might go with `mixed`.

Reply:
ok, makes sense ;)